plugins/aws-appmesh/config/netconfig.go

```diff
@@ -143,25 +143,42 @@ func validateConfig(config netConfigJSON) error {
 	}
 
 	// Validate the format of all fields.
+
+	if len(config.EgressIgnoredIPs) == 1 && config.EgressIgnoredIPs[0] == "" {
+		config.EgressIgnoredIPs = nil
+	}
+
 	if err := isValidPort(config.ProxyEgressPort); err != nil {
 		return err
 	}
 	if err := isValidPort(config.ProxyIngressPort); err != nil {
 		return err
 	}
 
+	//If incoming ports or ips are empty we still treat that as valid and delete that empty element
+	if len(config.AppPorts) == 1 && config.AppPorts[0] == "" {
+		config.AppPorts = nil
+	}
+
 	for _, port := range config.AppPorts {
 		if err := isValidPort(port); err != nil {
 			return err
 		}
 	}
 
+	if len(config.EgressIgnoredPorts) == 1 && config.EgressIgnoredPorts[0] == "" {
+		config.EgressIgnoredPorts = nil
+	}
+
 	for _, port := range config.EgressIgnoredPorts {
 		if err := isValidPort(port); err != nil {
 			return err
 		}
 	}
 
+	if len(config.EgressIgnoredPorts) == 1 && config.EgressIgnoredPorts[0] == "" {
+		config.EgressIgnoredPorts = nil
+	}
```

Comment:
why are we doing this block for EgressIgnoredPorts twice?

Reply:
updated